src/id128/id128.c

```diff
@@ -122,11 +123,23 @@ static int verb_show(int argc, char **argv, void *userdata) {
                         return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                "'show --app-specific=' can only be used with explicit UUID input.");
 
-                for (const GptPartitionType *e = gpt_partition_type_table; e->name; e++) {
-                        r = show_one(&table, e->name, e->uuid, e == gpt_partition_type_table);
+                if (arg_var_partition_uuid) {
+                        /* The DPS says that the UUID for /var/ should be keyed with machine-id. */
+                        sd_id128_t uuid;
+
+                        r = sd_id128_get_machine_app_specific(SD_GPT_VAR, &uuid);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to generate machine-specific UUID: %m");
+
+                        r = show_one(&table, "var-uuid", uuid, /* first= */ false);
```

Comment:
I'd really move this out of verb_show(). just handle this directly in run() and shortcut things there.